extension/src/popup/components/sendPayment/SendAmount/AssetSelect/index.tsx

```diff
@@ -6,13 +6,15 @@ import { ROUTES } from "popup/constants/routes";
 import { navigateTo } from "popup/helpers/navigate";
 import { AssetIcon } from "popup/components/account/AccountAssets";
 import { transactionSubmissionSelector } from "popup/ducks/transactionSubmission";
+import { useIsSwap } from "popup/helpers/useIsSwap";
 
 import "./styles.scss";
 
 export enum ASSET_SELECT {
   QUERY_PARAM = "assetSelect",
   SOURCE = "source",
   DEST = "dest",
+  SWAP_QUERY_PARAM = "isSwap",
```

Comment:
I think this is fine for this PR, but I'm wondering if it makes sense to move all this asset select stuff either into Redux or (perhaps more appropriate for this kind of ephemeral data) React Context.

Relying too much on query params always make me a little nervous as route names, etc. can change, so sometimes it's nicer to rely on something more deterministic. Wdyt? It could be done in a future bug ticket.

Reply:
That's a good point 🤔 it is getting a little too complex for using query params only. I can change to redux in this PR

Reply:
on second thought is a bigger change than I initially thought. I'll do this in a subsequent PR so it's not lost in this PR's changes 😄

Reply:
tracking here https://stellarorg.atlassian.net/browse/WAL-446